core/src/main/java/org/jobrunr/scheduling/ScheduleExpressionType.java

```diff
@@ -0,0 +1,36 @@
+package org.jobrunr.scheduling;
+
+import org.jobrunr.scheduling.cron.CronExpression;
+import org.jobrunr.scheduling.interval.Interval;
+import org.jobrunr.scheduling.interval.InvalidScheduleException;
+
+public enum ScheduleExpressionType {
+    CRON_EXPRESSION,
+    INTERVAL,
+    INVALID,
+    UNKNOWN;
+
+    public static ScheduleExpressionType getScheduleType(String scheduleExpression) {
+        if (scheduleExpression == null || scheduleExpression.isEmpty()) {
+            return INVALID;
+        } else if (scheduleExpression.matches(".*\\s.*")) {
+            return CRON_EXPRESSION;
+        } else if (scheduleExpression.startsWith("P")) {
+            return INTERVAL;
+        }
+
+        return UNKNOWN;
+    }
+
+    public static Schedule createSchedule(String scheduleExpression) {
+        ScheduleExpressionType type = ScheduleExpressionType.getScheduleType(scheduleExpression);
+
+        if (type.equals(ScheduleExpressionType.CRON_EXPRESSION)) {
+            return CronExpression.create(scheduleExpression);
+        } else if (type.equals(ScheduleExpressionType.INTERVAL)) {
+            return new Interval(scheduleExpression);
+        } else {
+            throw new InvalidScheduleException(scheduleExpression);
+        }
+    }
```

Comment:
I was more thinking something in the line of:

http://tutorials.jenkov.com/java/enums.html#enum-abstract-methods.

I also think we can get away with only invalid if it is null or empty. So I would remove unknown for now. So, this would become (pseudo code, not tested): 

```java
public enum ScheduleExpressionType {
    CRON_EXPRESSION {
        @Override
        public Schedule createSchedule() {
            return CronExpression.create(scheduleExpression);
        }
    },
    INTERVAL {
        @Override
        public Schedule createSchedule() {
            return new Interval(scheduleExpression);
        }
    },
    INVALID {
        @Override
        public Schedule createSchedule() {
            throw new InvalidScheduleException(scheduleExpression);
        }
    };

    public static ScheduleExpressionType getScheduleType(String scheduleExpression) {
        if (scheduleExpression.matches(".*\\s.*")) {
            return CRON_EXPRESSION;
        } else if (scheduleExpression.startsWith("P")) {
            return INTERVAL;
        }
        return INVALID;
    }

  public abstract Schedule createSchedule(String scheduleExpression);
}
```

Reply:
Oh, I see! I hope the new changes are more in line of what you were thinking then ^^